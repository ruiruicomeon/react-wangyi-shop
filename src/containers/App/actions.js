import { LOAD_APP } from './constants';

export function loadApp() {
  return {
    type: LOAD_APP,
  };
}
