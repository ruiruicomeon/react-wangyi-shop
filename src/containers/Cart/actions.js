import {
  ADD_TO_CART,
  DEL_FROM_CART,
  CHECK_CART,
  CHANGE_CART_QUANTITY,
  TRANSFER_TO_USER_CART,
  LOAD_CARTS,
  SET_CARTS,
  SET_IS_CHECK_ALL,
  SET_IS_CHECK_ONE,
  CHECK_ALL,
  UN_CHECK_ALL,
} from './constants';

export function addToCart(product) {
  return {
    type: ADD_TO_CART,
    payload: { product },
  };
}

export function changeCartQuantity(product) {
  return {
    type: CHANGE_CART_QUANTITY,
    payload: { product },
  };
}

export function checkCart(product) {
  return {
    type: CHECK_CART,
    payload: { product },
  };
}

export function delFromCart(product) {
  return {
    type: DEL_FROM_CART,
    payload: { product },
  };
}

export function transferToUserCart() {
  return {
    type: TRANSFER_TO_USER_CART,
  };
}

export function loadCarts() {
  return {
    type: LOAD_CARTS,
  };
}

export function setCarts(carts) {
  return {
    type: SET_CARTS,
    payload: { carts },
  };
}

export function setIsCheckAll(isCheckAll) {
  return {
    type: SET_IS_CHECK_ALL,
    payload: { isCheckAll },
  };
}

export function setIsCheckOne(isCheckOne) {
  return {
    type: SET_IS_CHECK_ONE,
    payload: { isCheckOne },
  };
}

export function checkAll() {
  return {
    type: CHECK_ALL,
  };
}

export function unCheckAll() {
  return {
    type: UN_CHECK_ALL,
  };
}
