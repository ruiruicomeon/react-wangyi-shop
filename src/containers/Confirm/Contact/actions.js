import { FETCH_CONTACTS, SET_CONTACTS, SAVE_CONTACT, SET_ERROR, SET_CONTACT, SET_IS_EDIT } from './constants';

export function fetchContacts() {
  return {
    type: FETCH_CONTACTS,
  };
}

export function setContacts(contacts) {
  return {
    type: SET_CONTACTS,
    payload: {
      contacts,
    },
  };
}

export function setError(error) {
  return {
    type: SET_ERROR,
    payload: {
      error,
    },
  };
}

export function saveContact(contact) {
  return {
    type: SAVE_CONTACT,
    payload: {
      contact,
    },
  };
}

export function setContact(contact) {
  return {
    type: SET_CONTACT,
    payload: {
      contact,
    },
  };
}

export function setIsEdit(isEdit) {
  return {
    type: SET_IS_EDIT,
    payload: {
      isEdit,
    },
  };
}
