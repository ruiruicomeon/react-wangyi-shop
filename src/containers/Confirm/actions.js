import { MAKE_ORDER } from './constants';

export function makeOrder() {
  return {
    type: MAKE_ORDER,
  };
}
