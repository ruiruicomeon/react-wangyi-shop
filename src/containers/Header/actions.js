import { FETCH_HOT_WORDS, SET_HOT_WORDS, FETCH_CATEGORIES, SET_CATEGORIES } from './constants';

export function fetchCategories() {
  return {
    type: FETCH_CATEGORIES,
  };
}

export function setCategories(categories) {
  return {
    type: SET_CATEGORIES,
    payload: {
      categories,
    },
  };
}

export function setHotWords(hotWords) {
  return {
    type: SET_HOT_WORDS,
    payload: {
      hotWords,
    },
  };
}

export function fetchHotWords() {
  return {
    type: FETCH_HOT_WORDS,
  };
}
