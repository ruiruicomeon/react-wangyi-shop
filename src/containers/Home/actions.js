import {
  FETCH_CAROUSELS,
  SET_CAROUSELS,
  FETCH_NEW_PRODUCTS,
  SET_NEW_PRODUCTS,
  FETCH_RECOMMEND_PRODUCTS,
  SET_RECOMMEND_PRODUCTS,
  FETCH_BEST_SELL_PRODUCTS,
  SET_BEST_SELL_PRODUCTS,
  FETCH_TIME_PRODUCTS,
  SET_TIME_PRODUCTS,
  FETCH_WELFARE_PRODUCTS,
  SET_WELFARE_PRODUCTS,
  FETCH_PRESENT_PRODUCTS,
  SET_PRESENT_PRODUCTS,
  FETCH_CATEGORY_LIST,
  SET_CATEGORY_LIST,
} from './constants';

export function fetchCarousels() {
  return {
    type: FETCH_CAROUSELS,
  };
}

export function setCarousels(carousels) {
  return {
    type: SET_CAROUSELS,
    payload: {
      carousels,
    },
  };
}

export function fetchNewProducts() {
  return {
    type: FETCH_NEW_PRODUCTS,
  };
}

export function setNewProducts(newProducts) {
  return {
    type: SET_NEW_PRODUCTS,
    payload: {
      newProducts,
    },
  };
}

export function fetchRecommendProducts() {
  return {
    type: FETCH_RECOMMEND_PRODUCTS,
  };
}

export function setRecommendProducts(recommendProducts) {
  return {
    type: SET_RECOMMEND_PRODUCTS,
    payload: {
      recommendProducts,
    },
  };
}

export function fetchBestSellProducts() {
  return {
    type: FETCH_BEST_SELL_PRODUCTS,
  };
}

export function setBestSellProducts(bestSellProducts) {
  return {
    type: SET_BEST_SELL_PRODUCTS,
    payload: {
      bestSellProducts,
    },
  };
}

export function fetchTimeProducts() {
  return {
    type: FETCH_TIME_PRODUCTS,
  };
}

export function setTimeProducts(timeProducts) {
  return {
    type: SET_TIME_PRODUCTS,
    payload: {
      timeProducts,
    },
  };
}

export function fetchWelfareProducts() {
  return {
    type: FETCH_WELFARE_PRODUCTS,
  };
}

export function setWelfareProducts(welfareProducts) {
  return {
    type: SET_WELFARE_PRODUCTS,
    payload: {
      welfareProducts,
    },
  };
}

export function fetchPresentProducts() {
  return {
    type: FETCH_PRESENT_PRODUCTS,
  };
}

export function setPresentProducts(presentProducts) {
  return {
    type: SET_PRESENT_PRODUCTS,
    payload: {
      presentProducts,
    },
  };
}

export function fetchCategoryList() {
  return {
    type: FETCH_CATEGORY_LIST,
  };
}

export function setCategoryList(categoryList) {
  return {
    type: SET_CATEGORY_LIST,
    payload: {
      categoryList,
    },
  };
}
