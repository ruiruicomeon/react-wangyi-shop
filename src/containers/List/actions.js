import { FETCH_CAROUSELS, SET_CAROUSELS, FETCH_PRODUCTS, SET_CATEGORY, SET_THIRD_CATEGORIES, SET_PRODUCTS } from './constants';

export function fetchCarousels(categoryId) {
  return {
    type: FETCH_CAROUSELS,
    payload: {
      categoryId,
    },
  };
}

export function setCarousels(carousels) {
  return {
    type: SET_CAROUSELS,
    payload: {
      carousels,
    },
  };
}

export function fetchProducts(categoryId) {
  return {
    type: FETCH_PRODUCTS,
    payload: {
      categoryId,
    },
  };
}

export function setCategory(category) {
  return {
    type: SET_CATEGORY,
    payload: {
      category,
    },
  };
}

export function setThirdCategories(thirdCategories) {
  return {
    type: SET_THIRD_CATEGORIES,
    payload: {
      thirdCategories,
    },
  };
}

export function setProducts(products) {
  return {
    type: SET_PRODUCTS,
    payload: {
      products,
    },
  };
}
