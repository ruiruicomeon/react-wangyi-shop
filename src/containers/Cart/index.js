import React from 'react';
import classnames from 'classnames';
import _ from 'lodash';
import { connect } from 'react-redux';
import { createStructuredSelector } from 'reselect';
import { push } from 'connected-react-router';

import { showLogin } from '@/containers/Login/actions';
import { makeSelectUser } from '@/containers/Login/selectors';
import { makeSelectCarts, makeSelectIsCheckAll, makeSelectIsCheckOne } from './selectors';
import { delFromCart, changeCartQuantity, checkCart } from '@/containers/Cart/actions';
import { checkAll, unCheckAll } from './actions';
import Loading from '@/components/Loading';
import Footer from '@/components/Footer';
import Checkbox from '@/components/Checkbox';
import Header from '@/containers/Header';
import Nav from '@/containers/Nav';
import CartItem from './CartItem';
import { getInfoOfSpecs } from '@/utils/libs';

import $style from './index.module.scss';

export class Cart extends React.Component {
  constructor(props) {
    super(props);

    this.state = {};
  }

  onDelete = product => {
    this.props.onDelFromCart(product);
  };

  onCheck = product => {
    this.props.onCheckCart(product);
  };

  onChangeQuantity = (product, quantity) => {
    product.quantity = quantity;
    this.props.onChangeCartQuantity(product);
  };

  onClickBuy = () => {
    if (!this.props.isCheckOne) {
      return;
    }

    if (_.isEmpty(this.props.user)) {
      this.props.onShowLogin();
    } else {
      this.props.onClickBuy();
    }
  };

  render() {
    const { carts, isCheckAll, onCheckAll, onUnCheckAll, isCheckOne } = this.props;

    let sumPrice = 0;
    if (!_.isEmpty(carts)) {
      sumPrice = carts.reduce((total, c) => {
        const {
          specs,
          productSpecs,
          productInfo: { prices },
        } = c;
        let price = getInfoOfSpecs(specs, productSpecs, prices);
        return total + price * c.quantity;
      }, 0);

      sumPrice = Math.round(sumPrice * 100) / 100;
    }

    return (
      <div className={$style.cart}>
        <Nav />
        <Header />
        <div className={classnames('container', $style.content)}>
          <div className={$style.header}>
            <Checkbox className={$style.header__checkbox} isChecked={isCheckAll} onCheck={() => (isCheckAll ? onUnCheckAll() : onCheckAll())} />
            <div className={$style.header__all}>全选</div>
            <div className={$style.header__info}>商品信息</div>
            <div className={$style.header__price}>单价</div>
            <div className={$style.header__quantity}>数量</div>
            <div className={$style.header__sum}>小计</div>
            <div className={$style.header__option}>操作</div>
          </div>
          <div className={$style.cart}>
            {_.isEmpty(carts) ? (
              <Loading />
            ) : (
              carts.map(p => {
                let id = p.id + p.specs.join('');
                return (
                  <CartItem
                    className={$style.cart__item}
                    key={id}
                    product={p}
                    onDelete={this.onDelete}
                    onCheck={this.onCheck}
                    onChangeQuantity={this.onChangeQuantity}
                  />
                );
              })
            )}
          </div>
          <div className={$style.footer}>
            <div className={$style.footer__left}>
              <Checkbox className={$style.footer__checkbox} isChecked={isCheckAll} onCheck={() => (isCheckAll ? onUnCheckAll() : onCheckAll())} />
              <div className={$style.footer__all}>全选</div>
            </div>
            <div className={$style.footer__right}>
              <div className={$style.footer__sum}>
                <div>应付总额：</div>
                <div className={$style.footer__price}>¥{sumPrice}</div>
              </div>
              <div
                className={classnames($style.footer__order, isCheckOne ? '' : $style.footer__order_disabled)}
                disabled={!isCheckOne}
                onClick={this.onClickBuy}
              >
                下单
              </div>
            </div>
          </div>
        </div>
        <Footer />
      </div>
    );
  }
}

const mapStateToProps = createStructuredSelector({
  carts: makeSelectCarts(),
  isCheckAll: makeSelectIsCheckAll(),
  isCheckOne: makeSelectIsCheckOne(),
  user: makeSelectUser(),
});

export function mapDispatchToProps(dispatch) {
  return {
    onDelFromCart: product => dispatch(delFromCart(product)),
    onCheckCart: product => dispatch(checkCart(product)),
    onChangeCartQuantity: product => dispatch(changeCartQuantity(product)),
    onCheckAll: () => dispatch(checkAll()),
    onUnCheckAll: () => dispatch(unCheckAll()),
    onClickBuy: () => dispatch(push('/confirm')),
    onShowLogin: () => dispatch(showLogin()),
  };
}

export default connect(mapStateToProps, mapDispatchToProps)(Cart);
