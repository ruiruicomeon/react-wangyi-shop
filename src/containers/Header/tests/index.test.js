import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { ConnectedRouter } from 'connected-react-router';

import history from '@/utils/history';
import configureStore from '@/store';
import { mapDispatchToProps } from '../index';
import { fetchHotWords, fetchCategories } from '../actions';
import { delFromCart } from '@/containers/Cart/actions';
import { Header } from '../index';

describe('Header 测试', () => {
  let store;

  beforeAll(() => {
    store = configureStore();
  });

  it('snapshot', () => {
    const loadSpy = jest.fn();
    const tree = renderer
      .create(
        <Provider store={store}>
          <ConnectedRouter history={history}>
            <Header hotWords={[]} categories={[]} carts={[]} onLoad={loadSpy} />
          </ConnectedRouter>
        </Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  describe('dispatch', () => {
    it('调用 onLoad 时 dispath fetchHotWords、fetchCategories', () => {
      const dispatch = jest.fn();
      const result = mapDispatchToProps(dispatch);
      result.onLoad();
      expect(dispatch).toHaveBeenCalledWith(fetchHotWords());
      expect(dispatch).toHaveBeenCalledWith(fetchCategories());
    });

    it('调用 onDelFromCart 时 dispath delFromCart', () => {
      const dispatch = jest.fn();
      const result = mapDispatchToProps(dispatch);
      const product = {
        id: 1,
        title: 'product',
      };
      result.onDelFromCart(product);
      expect(dispatch).toHaveBeenCalledWith(delFromCart(product));
    });
  });
});
