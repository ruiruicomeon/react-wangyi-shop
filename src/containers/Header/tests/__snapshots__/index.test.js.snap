// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header 测试 snapshot 1`] = `
<header
  className="header"
>
  <div
    className=""
  >
    <div
      className="container content"
    >
      <a
        className="logo"
        href="/"
        onClick={[Function]}
      >
        <span
          className="logo__img"
        />
      </a>
      <a
        className="smlogo"
        href="/"
        onClick={[Function]}
      >
        <span
          className="smlogo__img"
        />
      </a>
      <a
        className="cart ant-dropdown-trigger"
        href="/cart"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      >
        <span
          className="cart__icon"
        />
        <span
          className="cart__title"
        >
          购物车
        </span>
        <span
          className="cart__num"
        >
          0
        </span>
      </a>
      <a
        className="user"
        href="/user"
        onClick={[Function]}
      >
        <span
          className="user__icon"
        />
      </a>
      <div
        className="search"
      >
        <div
          className="search__main"
        >
          <div
            className="search__wrap"
          >
            <div
              className="search__hide"
            >
              &gt;&gt;
            </div>
            <div
              className="search__prefix"
            />
            <input
              className="search__input"
              onChange={[Function]}
              placeholder="搜索"
              type="text"
              value=""
            />
          </div>
          <div
            className="search__btn"
          >
            <div
              className="search__icon"
            />
            <div
              className="search__text"
              onClick={[Function]}
            >
              搜索
            </div>
          </div>
        </div>
      </div>
      <div
        className="loading"
      >
        <div
          className="ant-spin ant-spin-spinning"
        >
          <span
            aria-label="loading"
            className="anticon anticon-loading ant-spin-dot"
            role="img"
            style={
              Object {
                "fontSize": 24,
              }
            }
          >
            <svg
              aria-hidden="true"
              className="anticon-spin"
              data-icon="loading"
              fill="currentColor"
              focusable="false"
              height="1em"
              viewBox="0 0 1024 1024"
              width="1em"
            >
              <path
                d="M988 548c-19.9 0-36-16.1-36-36 0-59.4-11.6-117-34.6-171.3a440.45 440.45 0 00-94.3-139.9 437.71 437.71 0 00-139.9-94.3C629 83.6 571.4 72 512 72c-19.9 0-36-16.1-36-36s16.1-36 36-36c69.1 0 136.2 13.5 199.3 40.3C772.3 66 827 103 874 150c47 47 83.9 101.8 109.7 162.7 26.7 63.1 40.2 130.2 40.2 199.3.1 19.9-16 36-35.9 36z"
              />
            </svg>
          </span>
        </div>
      </div>
    </div>
  </div>
</header>
`;
